src/getPullChanges.ts

```diff
@@ -0,0 +1,49 @@
+import * as github from '@actions/github';
+import * as core from '@actions/core';
+import { SummaryModes } from './summaryModes.js'
+
+export async function getPullChanges(summaryMode: SummaryModes): Promise<String[]> {
+  // Skip Changes collection if we don't need it
+  if (summaryMode === SummaryModes.Mixed || summaryMode === SummaryModes.None) {
+    return [];
+  }
+
+  // Skip Changes collection if we can't do it
+  if (!['pull_request', 'pull_request_review', 'pull_request_review_comment'].includes(github.context.eventName)) {
+    return [];
+  }
+
+  const gitHubToken = core.getInput('github-token').trim();
+  const prNumber = github.context.payload.pull_request.number
+  try {
+    const client = new github.GitHub(gitHubToken)

```

Comment:
You'll have to create the octokit-client like this:

```suggestion
    const octokit: Octokit = github.getOctokit(gitHubToken);
```